plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TransactionLogAccess.java

```diff
@@ -269,8 +288,14 @@ private List<AddFileEntry> loadActiveFiles(TableSnapshot tableSnapshot, Metadata
                 typeManager,
                 fileSystemFactory.create(session),
                 fileFormatDataSourceStats,
-                Optional.of(new MetadataAndProtocolEntry(metadataEntry, protocolEntry)))) {
-            return activeAddEntries(checkpointEntries, transactions)
+                Optional.of(new MetadataAndProtocolEntry(metadataEntry, protocolEntry)),
+                partitionConstraint)) {
+            Stream<AddFileEntry> addFileEntryStream = activeAddEntries(checkpointEntries, transactions);
+            if (partitionConstraint.isAll()) {
+                return addFileEntryStream.collect(toImmutableList());
```

Comment:
nit: this can be written more tersely 
```
.filter(partitionConstraint.isAll()
                            ? addAction -> true
                            : addAction -> partitionMatchesPredicate(...)
```